src/strings_internal.h

```diff
@@ -92,8 +91,10 @@ class StringParameters {
 	template <typename T>
 	T GetNextParameter()
 	{
-		auto ptr = GetNextParameterPointer();
-		return static_cast<T>(ptr->data);
+		const auto &param = GetNextParameterReference();
+		/* Check that we hold a uint64_t. std::get_if is used as std::get would check again. */
+		if (std::holds_alternative<uint64_t>(param.data)) return static_cast<T>(*std::get_if<uint64_t>(&param.data));
+		throw std::runtime_error("Attempt to read string parameter as integer");
```

Comment:
Doesn't `std::get_if` implicitly check again, because *if* the type is wrong it will return `nullptr`. As such, it would be doing some checks to determine whether to return `nullptr`. Maybe something like the following would be simpler:
```suggestion
		uint64_t *data = std::get_if<uint64_t>(&param.data);
		if (data != nullptr) return static_cast<T>(*data);
		throw std::runtime_error("Attempt to read string parameter as integer");
```